scripts/quests/sandoria/PLD_AF1_Sharpening_The_Sword.lua

```diff
@@ -0,0 +1,160 @@
+                    elseif player:getCharVar("Quest[0][90]Prog") == 1 then --declined first
+                        return quest:event(43)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [45] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 2)
+                    else
+                        quest:setVar(player, 'Prog', 1) -- Declined First offering.
+                    end
+                end,
+
+                [43] = function(player, csid, option, npc)
+                    if option == 1 then -- Accept quest.
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 2)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Ailbeche'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.ORDELLE_WHETSTONE) then
+                        return quest:progressEvent(44)
+                    else
+                        return quest:event(42)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [44] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.ORDELLE_WHETSTONE)
+                    end
+                end,
+            },
+
+        },
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Sobane'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') >= 2 then
+                        return quest:progressEvent(52) -- Finish quest.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [52] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+
+        [xi.zone.ORDELLES_CAVES] =
+        {
+            ['Stalagmite'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress == 3 and
+                        quest:getVar(player, 'PolevikKilled') == 0
+                    then
+                        if npcUtil.popFromQM(player, npc, ordellesID.mob.POLEVIK, { claim = true, hide = 0 }) then
+                            player:messageSpecial(ordellesID.text.SENSE_OF_FOREBODING)
```

Comment:
need to return this as a quest message

Reply:
ive found that it will still work without the return but its there now ;)